foxpuppet/tests/test_browser_model.py

```diff
@@ -0,0 +1,39 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from foxpuppet import FoxPuppet
+
+
+class TestBrowserModel(object):
+
+    def test_new_private_window(self, selenium):
+        """Tests opening a new private browsing window via menu"""
+        foxpuppet = FoxPuppet(selenium)
+        with foxpuppet.windows.wait_for_new_window():
```

Comment:
This should really be done transparently. If you have to to do it each time when you open a window, it will blow up the tests horribly. Why don't you pick the code from puppeteer's windows.py module?